WordPress/gutenberg
# Add media query hooks and animation to snackbar notices

This PR adds some animation to snackbar notices. It uses `react-spring` to do so and it introduces new React hooks: `useMediaQuery` and `useReducedMotion` in order to disable the animation if the reduced motion preference is enabled.

**Testing instructions**

 - Open the console
 - Add a notice

```js
wp.data.dispatch('core/notices').createNotice(
		'info',
		'Post published.',
		{
			type: 'snackbar',
			actions: [
				{
					onClick: () => {},
					label: 'View post'
				}
			]
		}
	);
```
